---
title: "REU 2020 Summer Internship"
author: "Khaleb Enoch"
date: "August 13, 2020"
output: pdf_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## This Projects Puropose
The REU summer project is about understanding brood parasites and how these parasites have evolved their unique method of using similiar 
species to take care of their offspring. This project is looking at sex chromosme evolution as the possible method for this parenting 
behavior "arising". In order to study this evolution, the W chromsome of these organisms need to be reconstructed.

### What We Did

## Mapping back to reference genome
First thing done was map the sequenced reads back to the reference genome, to calculate coverage of each scaffold in the samples.
# Load up bwa package, index the reference genome, and combine the forward and reverse reads into one sam file.
```{bash, run=F}
module load bwa/0.7.17-fasrc01
bwa index -p netAur_index <Reference Genome>.fna.gz
bwa mem -t 8 netAur_index <File Name R1>.fastq.gz <File Name R2>.fastq.gz > <Output File>.sam
```
## Calculate the coverage of scaffolds

## Use samtools to clean up files

## Filter coverage output to find autosomes and sex chromosomes
#Install and load required packages. (tidyverse,rpatchwork,Rcolorbrewer,ggplot2(included in tidyverse)
```{r}
install.packages(<Package Name>)
library(<Package Name>)
```

## Create bed file from filter autosomes and sex chromsomes
Created Bed files containing autosomes and possible W chromosomes for each sex and ran samtools depth on them.

## Run samtools depth on the bed files to get coverage for each mapped read

## Compare and graph autosomal and sex chromosome data between male and females
add 'male autosomes vs female autosomes' and 'male W sex chromosomes vs female w sex chromsomes' plots

## Confirm Identitiy of scaffolds by comparing coverage levels in males and females
add 'male and female sex chromsome' and 'male vs female autosome' plots

## Normalize Meandepth
add plots of 'meandepth vs endpos' and 'normdp vs endopos' and 'normdp.male vs normdp.female'

## Using normdp male vs female plot, assign scaffolds to a location
add filtered plot of normdp male vs female and filtered histogram
```{r}
ggplot(NA_both_filtered, aes(x=rel.dp, col=chr))+ geom_histogram(bins=50)
ggplot(NA_both_filtered, aes(x=rel.dp,y=normdp.female, col=chr))+ geom_point()
ggplot(NA_both_filtered, aes(x=normdp.male,y=normdp.female, col=chr))+ geom_point()
```
## Challenge our results
add plot of 'coverage in male vs female'
```{r,include=FALSE}
NA_both_filtered %>% filter(chr == 'W') %>% ggplot(aes(x=coverage.female, y=coverage.male))+ geom_point() #expect 100 coverage in fem, and expect lower coverage in males
```
## Confirm results by blasting W scaffolds to chicken genome
